### ISSUE TITLE
[SCM-981] Several integration tests are never run and fail if you do

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-scm#1211]
**[Niels Basjes](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nielsbasjes)** opened **[SCM-981](https://issues.apache.org/jira/browse/SCM-981?redirect=false)** and commented

The README.md states

> Run all the tests with `mvn -Prun-its verify` to assure nothing else was accidentally broken.

The Github actions reuses for the CI build

```java
apache/maven-gh-actions-shared/.github/workflows/maven-verify.yml@v2
```

which does the same

```java
./mvnw --errors --batch-mode --show-version -D"invoker.streamLogsOnFailures" -P run-its verify
```

https://github.com/apache/maven-gh-actions-shared/blob/v2/.github/workflows/maven-verify.yml#L167
https://github.com/apache/maven-gh-actions-shared/blob/v2/.github/workflows/maven-verify.yml#L93

The snag is that this profile does not exist in this project.

I asked this and was told to use the `tck-git` profile
https://github.com/apache/maven-scm/pull/134#issuecomment-1130214768

I found that there is also a `tck-svn` profile and a `tck` profile.

When I run `mvn clean verify -Ptck` there are several tests that fail and are apparently never run by anyone.

 

Proposal: 
* Create a profile run-its that simply activates the appropriate tck profiles.
* Fix the failing tests.

 

---

**Remote Links:**
- [GitHub Pull Request #140
](https://github.com/apache/maven-scm/pull/140)

**Comment by jira-importer** [comment]
**[Niels Basjes](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nielsbasjes)** commented

Tests I know about that fail under `-Ptck`
* maven-scm-providers/maven-scm-provider-hg/src/test/java/org/apache/maven/scm/provider/hg/command/blame/HgBlameCommandTckTest.java
  This is never true; the revision is always a hash.

```java
assertEquals( "0", line.getRevision() ); 
```

* maven-scm-providers/maven-scm-provider-local/src/test/java/org/apache/maven/scm/provider/local/command/update/LocalUpdateCommandTckTest.java
  In the testUpdateCommand test located in the superclass maven-scm-test/src/main/java/org/apache/maven/scm/tck/command/update/UpdateCommandTckTest.java fails over 

```java
assertPath( "src/main/java/org/Foo.java", file.getPath() );
```

with

```java
Expected :src/main/java/org/Foo.java
Actual   :/src/main/java/org/Foo.java
```

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

This, I need to tackle for the next release...

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [58e35c3dffa1d39b0dcae7c791e27bfc7c7f700b](https://gitbox.apache.org/repos/asf?p=maven-scm.git;a=commit;h=58e35c3dffa1d39b0dcae7c791e27bfc7c7f700b).
